### PR TITLE
dts: renesas_ra: Changing to describe clocks with DeviceTree's standard manner

### PR DESCRIPTION
**Pull request by soburi**
DeviceTree typically references the clock source using the `clocks`
property defined in `base.yaml`, so we'll change it to this.

Move the process of replacing numerical values with macros to
the header, and set the division ratio in a numeric without
using macros in the device tree.

### CODE REVIEW

File: include/zephyr/devicetree.h [hunk 1]
```diff
@@ -523,6 +523,92 @@
...
+ *
+ * @code{.c}
+ *    DT_NODE_FULL_NAME_UNQUOTED(DT_NODELABEL(node)) // my-node@12345678
+ * @endcode
+ *
+ * @param node_id node identifier
+ * @return the node's full name with unit-address as a sequence of tokens,
+ *         with no quotes
+ */
+#define DT_NODE_FULL_NAME_UNQUOTED(node_id) DT_CAT(node_id, _FULL_NAME_UNQUOTED)
```
Comment (gmarull) [inline include/zephyr/devicetree.h:552]: add tests and macros.bnf if needed.

Comment (soburi) [reply include/zephyr/devicetree.h:552]: Thank you for the review. I forgot about it. I've added it, so please check it again.

**Review by gmarull — DISMISSED** [review]

**Review by duynguyenxa — DISMISSED** [review]
Thank you for the implementation

**Comment by soburi** [comment]
@duynguyenxa 

> Thank you for the implementation

Thank you for your review.

Please check also https://github.com/zephyrproject-rtos/hal_renesas/pull/36.

**Comment by soburi** [comment]
This needs to be merged before https://github.com/zephyrproject-rtos/zephyr/pull/78959 (or any other PR that includes hal-renesas updating).

**Comment by soburi** [comment]
@gmarull @duynguyenxa 

The hal module has been updated so it has been rebased.
Could you check again please?

**Review by gmarull — APPROVED** [review]

**Review by duynguyenxa — APPROVED** [review]

**Comment by soburi** [comment]
@nordic-krch 
Could you take a look?

**Review by nordic-krch — APPROVED** [review]

**Pull request merged**